src/ray/core_worker/core_worker.cc

```diff
@@ -2301,24 +2312,11 @@ void CoreWorker::PlasmaCallback(SetResultCallback success,
       it->second.push_back({plasma_arrived_callback});
     }
   }
-  SubscribeToPlasmaAdd(object_id);
 
-  // Check in-memory store in case object became ready *before* SubscribeToPlasmaAdd.
-  if (Get(std::vector<ObjectID>{object_id}, 0, &vec).ok() && vec.size() > 0) {
-    std::vector<std::function<void(void)>> callbacks;
-    {
-      absl::MutexLock lock(&plasma_mutex_);
-      auto after_iter = async_plasma_callbacks_.extract(object_id);
-      callbacks = after_iter.mapped();
-    }
-    for (auto callback : callbacks) {
-      // This callback needs to be asynchronous because it runs on the io_service_, so no
-      // RPCs can be processed while it's running. This can easily lead to deadlock (for
-      // example if the callback calls ray.get() on an object that is dependent on an RPC
-      // to be ready).
-      callback();
-    }
-  }
+  // Ask raylet to subscribe to object notification. Raylet will call this core worker
+  // when the object is local. HandlePlasmaObjectReady handles such request.
```

Comment:
Add a comment that the raylet will fire the callback immediately if the object is already available